onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py

```diff
@@ -225,6 +224,216 @@ def __init__(
+
+        inputs = {
+            "input_ids": input_ids.contiguous(),
+            "attention_mask": attention_mask.contiguous(),
+        }
+
+        if add_position_ids:
+            position_ids = attention_mask.long().cumsum(-1) - 1
+            position_ids.masked_fill_(attention_mask == 0, 1)
+            inputs["position_ids"] = position_ids.contiguous()
+
+        if add_past_kv_inputs:
+            torch_dtype = torch.float16 if use_fp16 else torch.float32
+            batch_size, sequence_length = input_ids.shape
+            max_sequence_length = config.max_position_embeddings
+            num_heads, head_size = (
+                config.num_key_value_heads,
+                config.hidden_size // config.num_attention_heads,
+            )
+            for i in range(config.num_hidden_layers):
+                past_key = torch.zeros(
+                    batch_size,
+                    num_heads,
+                    max_sequence_length if use_buffer_share else 0,
+                    head_size,
+                    device=device,
+                    dtype=torch_dtype,
+                )
+                past_value = torch.zeros(
+                    batch_size,
+                    num_heads,
+                    max_sequence_length if use_buffer_share else 0,
+                    head_size,
+                    device=device,
+                    dtype=torch_dtype,
+                )
+                inputs.update(
+                    {
+                        f"past_key_values.{i}.key": past_key.contiguous(),
+                        f"past_key_values.{i}.value": past_value.contiguous(),
+                    }
+                )
+
+        return inputs
+
+    def get_calib_inputs(self, dataset_name, model_name, cache_dir, calib_size, batch_size, block_size, device, use_fp16,
+                        use_buffer_share, add_past_kv_inputs, max_calib_rows_to_load, add_position_ids):
+        # Access transformers and datasets from the instance variables
+        AutoConfig = self.AutoConfig
+        AutoTokenizer = self.AutoTokenizer
+        load_dataset = self.load_dataset
+
+        config = AutoConfig.from_pretrained(model_name, use_auth_token=True, cache_dir=cache_dir,
+                                           trust_remote_code=True)
+        tokenizer = AutoTokenizer.from_pretrained(model_name, use_auth_token=True, cache_dir=cache_dir,
+                                                 trust_remote_code=True)
+        tokenizer.add_special_tokens({'pad_token': '[PAD]'})
+        tokenizer.pad_token = tokenizer.eos_token
+
+        assert calib_size <= max_calib_rows_to_load, "calib size should be no more than max_calib_rows_to_load"
+
+        if "cnn" in dataset_name:
+            dataset2 = load_dataset("cnn_dailymail", name="3.0.0", split="train").select(range(max_calib_rows_to_load))
+            column = "article"
+        elif "pilevel" in dataset_name:
```

Comment:
why the name is pilevel? It is pile validation set. Suggest to to use name "pile" or "pileval"

Reply:
Done